### PR TITLE
Refactored list component to composition API

### LINKED ISSUE #540: Refactor ListComponent

### PR DESCRIPTION
**Pull request by te4-adrian-almetun-smeds**
## ⚙️ What issue did you implement or fix?
Fixes #540

## 📜 Summary
- Refactored to composition API

## 📝 Type of change

- [x] ✨ New feature or enhancement (non-breaking change which adds functionality)

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: frontend/src/components/ListComponent.vue [hunk 1]
```diff
@@ -36,81 +36,101 @@
 
 <script lang="ts">
 import { Component, Prop, Vue } from 'vue-property-decorator';
```
Comment (te4-alexander-kjellberg1) [inline frontend/src/components/ListComponent.vue]: is "import { Component, Prop, Vue } from 'vue-property-decorator';" needed?

Comment (te4-adrian-almetun-smeds) [reply frontend/src/components/ListComponent.vue]: No, it's not. It's been removed.

**Review by TE4-David-Sundqvist — APPROVED** [review]
👍 looks great!

**Review by te4-andre-skvarc — APPROVED** [review]
Nice, I like the comments. Makes it very understandable!

**Pull request merged**